src/Microsoft.Azure.SignalR.Protocols/ServiceProtocol.cs

```diff
@@ -439,6 +453,22 @@ private static HandshakeResponseMessage CreateHandshakeResponseMessage(byte[] in
             return new HandshakeResponseMessage(errorMessage);
         }
 
+        private static PingMessage CreatePingMessage(byte[] input, ref int offset, int arrayLength)
+        {
+            if (arrayLength >= 2)
+            {
+                var length = ReadMapLength(input, ref offset, "messages");
+                var dict = new Dictionary<string, string>((int)length);
```

Comment:
This is extremely inefficient. We're making a new dictionary per ping? Can we do something else here? Either reuse a single dictionary or come up with a more compact format. 

Maybe we need some performance tests to make sure we don't regress basic scenarios.

Reply:
@davidfowl do you means message pack format?

Reply:
Yes an array would be better.Also maybe we could allocate it once and reuse it